bayes_opt/logger.py

```diff
@@ -155,18 +154,16 @@ def _step(self, instance, colour=_colour_regular_message):
         A stringified, formatted version of the most recent optimization step.
         """
         res = instance.res[-1]
-        cells = []
+        cells: list[str | None] = [None, None, None, None]
```

Comment:
Maybe a tuple would be a more appropriate data structure here?

Reply:
I prefer `tuple`. But since `tuple` is immutable, I used `list`.

If I were to change to `tuple`, it would require a lot of changes in the underlying logic.

Reply:
you could do 
```
num_cells = 4
cells: list[str|None] = [None] * num_cells
```
(or just `[None] * 4`)

Reply:
> you could do
> 
> ```
> num_cells = 4
> cells: list[str|None] = [None] * num_cells
> ```
> 
> (or just `[None] * 4`)

Is `[None] * 4` more efficient than `[None, None, None, None]`?
If it is more efficient, then it is correct to modify it.

Reply:
I don't think it's more efficient, if it's performance critical code, leave it as is.
Just a bit less repetition and more flexible if there ever could be different numbers of cells